Crash in nested settings screens

Originally detected while reviewing #7792, but not directly related, confirmed the bug exists in `develop` as well.

### Expected behavior
App should not crash when a nested settings screen is open and user switches to another app, then back to WP android app.

### Actual behavior
With the dev options "re-create activities" set to ON, open any of the app's nested settings screen, then switch to another app, then switch back to WP Android.

```
05-22 12:39:44.068 19845-19845/org.wordpress.android V/FA: onActivityCreated
05-22 12:39:44.090 19845-19850/org.wordpress.android I/zygote64: Do full code cache collection, code=1002KB, data=657KB
05-22 12:39:44.092 19845-19850/org.wordpress.android I/zygote64: After code cache collection, code=930KB, data=542KB
05-22 12:39:44.140 19845-19845/org.wordpress.android E/Parcel: Class not found when unmarshalling: android.support.v7.widget.Toolbar$SavedState
                                                               java.lang.ClassNotFoundException: android.support.v7.widget.Toolbar$SavedState
                                                                   at java.lang.Class.classForName(Native Method)
                                                                   at java.lang.Class.forName(Class.java:453)
                                                                   at android.os.Parcel.readParcelableCreator(Parcel.java:2888)
                                                                   at android.os.Parcel.readParcelable(Parcel.java:2842)
                                                                   at android.os.Parcel.readValue(Parcel.java:2745)
                                                                   at android.os.Parcel.readSparseArrayInternal(Parcel.java:3195)
                                                                   at android.os.Parcel.readSparseArray(Parcel.java:2407)
                                                                   at android.os.Parcel.readValue(Parcel.java:2802)
                                                                   at android.os.Parcel.readArrayMapInternal(Parcel.java:3114)
                                                                   at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:273)
                                                                   at android.os.BaseBundle.unparcel(BaseBundle.java:226)
                                                                   at android.os.Bundle.getSparseParcelableArray(Bundle.java:1009)
                                                                   at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2129)
                                                                   at android.app.Dialog.onRestoreInstanceState(Dialog.java:470)
                                                                   at android.preference.PreferenceScreen.showDialog(PreferenceScreen.java:217)
                                                                   at android.preference.PreferenceScreen.onRestoreInstanceState(PreferenceScreen.java:282)
                                                                   at android.preference.Preference.dispatchRestoreInstanceState(Preference.java:2035)
                                                                   at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:324)
                                                                   at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:329)
                                                                   at android.preference.Preference.restoreHierarchyState(Preference.java:2017)
                                                                   at android.preference.PreferenceFragment.onActivityCreated(PreferenceFragment.java:218)
                                                                   at org.wordpress.android.ui.prefs.AppSettingsFragment.onActivityCreated(AppSettingsFragment.java:174)
                                                                   at android.app.Fragment.performActivityCreated(Fragment.java:2517)
                                                                   at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1299)
                                                                   at android.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1557)
                                                                   at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1618)
                                                                   at android.app.FragmentManagerImpl.dispatchMoveToState(FragmentManager.java:3027)
                                                                   at android.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2979)
                                                                   at android.app.FragmentController.dispatchActivityCreated(FragmentController.java:178)
                                                                   at android.app.Activity.performCreate(Activity.java:7005)
                                                                   at android.app.Activity.performCreate(Activity.java:6990)
                                                                   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
                                                                   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731)
                                                                   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
                                                                   at android.app.ActivityThread.-wrap11(Unknown Source:0)
                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
                                                                   at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                   at android.os.Looper.loop(Looper.java:164)
                                                                   at android.app.ActivityThread.main(ActivityThread.java:6494)
                                                                   at java.lang.reflect.Method.invoke(Native Method)
                                                                   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
                                                                Caused by: java.lang.ClassNotFoundException: android.support.v7.widget.Toolbar$SavedState
                                                                   at java.lang.Class.classForName(Native Method)
                                                                   at java.lang.BootClassLoader.findClass(ClassLoader.java:1355)
                                                                   at java.lang.BootClassLoader.loadClass(ClassLoader.java:1415)
                                                                   at java.lang.ClassLoader.loadClass(ClassLoader.java:312)
                                                                   at java.lang.Class.classForName(Native Method) 
                                                                   at java.lang.Class.forName(Class.java:453) 
                                                                   at android.os.Parcel.readParcelableCreator(Parcel.java:2888) 
                                                                   at android.os.Parcel.readParcelable(Parcel.java:2842) 
                                                                   at android.os.Parcel.readValue(Parcel.java:2745) 
                                                                   at android.os.Parcel.readSparseArrayInternal(Parcel.java:3195) 
                                                                   at android.os.Parcel.readSparseArray(Parcel.java:2407) 
                                                                   at android.os.Parcel.readValue(Parcel.java:2802) 
                                                                   at android.os.Parcel.readArrayMapInternal(Parcel.java:3114) 
                                                                   at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:273) 
                                                                   at android.os.BaseBundle.unparcel(BaseBundle.java:226) 
                                                                   at android.os.Bundle.getSparseParcelableArray(Bundle.java:1009) 
                                                                   at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2129) 
                                                                   at android.app.Dialog.onRestoreInstanceState(Dialog.java:470) 
                                                                   at android.preference.PreferenceScreen.showDialog(PreferenceScreen.java:217) 
                                                                   at android.preference.PreferenceScreen.onRestoreInstanceState(PreferenceScreen.java:282) 
                                                                   at android.preference.Preference.dispatchRestoreInstanceState(Preference.java:2035) 
                                                                   at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:324) 
                                                                   at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:329) 
                                                                   at android.preference.Preference.restoreHierarchyState(Preference.java:2017) 
                                                                   at android.preference.PreferenceFragment.onActivityCreated(PreferenceFragment.java:218) 
                                                                   at org.wordpress.android.ui.prefs.AppSettingsFragment.onActivityCreated(AppSettingsFragment.java:174) 
                                                                   at android.app.Fragment.performActivityCreated(Fragment.java:2517) 
                                                                   at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1299) 
                                                                   at android.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1557) 
                                                                   at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1618) 
                                                                   at android.app.FragmentManagerImpl.dispatchMoveToState(FragmentManager.java:3027) 
                                                                   at android.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2979) 
                                                                   at android.app.FragmentController.dispatchActivityCreated(FragmentController.java:178) 
                                                                   at android.app.Activity.performCreate(Activity.java:7005) 
                                                                   at android.app.Activity.performCreate(Activity.java:6990) 
                                                                   at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214) 
                                                                   at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731) 
                                                                   at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856) 
                                                                   at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                                   at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589) 
                                                                   at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                   at android.os.Looper.loop(Looper.java:164) 
                                                                   at android.app.ActivityThread.main(ActivityThread.java:6494) 
                                                                   at java.lang.reflect.Method.invoke(Native Method) 
                                                                   at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
                                                                   at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
                                                                Caused by: java.lang.NoClassDefFoundError: Class not found using the boot class loader; no stack trace available
05-22 12:39:44.140 19845-19845/org.wordpress.android D/AndroidRuntime: Shutting down VM
05-22 12:39:44.143 19845-19845/org.wordpress.android E/AndroidRuntime: FATAL EXCEPTION: main
                                                                       Process: org.wordpress.android, PID: 19845
                                                                       java.lang.RuntimeException: Unable to start activity ComponentInfo{org.wordpress.android/org.wordpress.android.ui.prefs.AppSettingsActivity}: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.v7.widget.Toolbar$SavedState
                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2778)
                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856)
                                                                           at android.app.ActivityThread.-wrap11(Unknown Source:0)
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589)
                                                                           at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                           at android.os.Looper.loop(Looper.java:164)
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6494)
                                                                           at java.lang.reflect.Method.invoke(Native Method)
                                                                           at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
                                                                        Caused by: android.os.BadParcelableException: ClassNotFoundException when unmarshalling: android.support.v7.widget.Toolbar$SavedState
                                                                           at android.os.Parcel.readParcelableCreator(Parcel.java:2916)
                                                                           at android.os.Parcel.readParcelable(Parcel.java:2842)
                                                                           at android.os.Parcel.readValue(Parcel.java:2745)
                                                                           at android.os.Parcel.readSparseArrayInternal(Parcel.java:3195)
                                                                           at android.os.Parcel.readSparseArray(Parcel.java:2407)
                                                                           at android.os.Parcel.readValue(Parcel.java:2802)
                                                                           at android.os.Parcel.readArrayMapInternal(Parcel.java:3114)
                                                                           at android.os.BaseBundle.initializeFromParcelLocked(BaseBundle.java:273)
                                                                           at android.os.BaseBundle.unparcel(BaseBundle.java:226)
                                                                           at android.os.Bundle.getSparseParcelableArray(Bundle.java:1009)
                                                                           at com.android.internal.policy.PhoneWindow.restoreHierarchyState(PhoneWindow.java:2129)
                                                                           at android.app.Dialog.onRestoreInstanceState(Dialog.java:470)
                                                                           at android.preference.PreferenceScreen.showDialog(PreferenceScreen.java:217)
                                                                           at android.preference.PreferenceScreen.onRestoreInstanceState(PreferenceScreen.java:282)
                                                                           at android.preference.Preference.dispatchRestoreInstanceState(Preference.java:2035)
                                                                           at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:324)
                                                                           at android.preference.PreferenceGroup.dispatchRestoreInstanceState(PreferenceGroup.java:329)
                                                                           at android.preference.Preference.restoreHierarchyState(Preference.java:2017)
                                                                           at android.preference.PreferenceFragment.onActivityCreated(PreferenceFragment.java:218)
                                                                           at org.wordpress.android.ui.prefs.AppSettingsFragment.onActivityCreated(AppSettingsFragment.java:174)
                                                                           at android.app.Fragment.performActivityCreated(Fragment.java:2517)
                                                                           at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1299)
                                                                           at android.app.FragmentManagerImpl.moveFragmentToExpectedState(FragmentManager.java:1557)
                                                                           at android.app.FragmentManagerImpl.moveToState(FragmentManager.java:1618)
                                                                           at android.app.FragmentManagerImpl.dispatchMoveToState(FragmentManager.java:3027)
                                                                           at android.app.FragmentManagerImpl.dispatchActivityCreated(FragmentManager.java:2979)
                                                                           at android.app.FragmentController.dispatchActivityCreated(FragmentController.java:178)
                                                                           at android.app.Activity.performCreate(Activity.java:7005)
                                                                           at android.app.Activity.performCreate(Activity.java:6990)
                                                                           at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1214)
                                                                           at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2731)
                                                                           at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2856) 
                                                                           at android.app.ActivityThread.-wrap11(Unknown Source:0) 
                                                                           at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1589) 
                                                                           at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                           at android.os.Looper.loop(Looper.java:164) 
                                                                           at android.app.ActivityThread.main(ActivityThread.java:6494) 
                                                                           at java.lang.reflect.Method.invoke(Native Method) 
                                                                           at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
                                                                           at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 
05-22 12:39:44.154 873-1664/? W/ActivityManager:   Force finishing activity org.wordpress.android/.ui.prefs.AppSettingsActivity
05-22 12:39:44.162 873-903/? I/ActivityManager: Showing crash dialog for package org.wordpress.android u0
```

### Steps to reproduce the behavior
1. make sure the dev options "re-create activities" is set to ON
2. open any of the app's nested settings screen, for example:
a) app settings -> notification settings -> pick a site so a nested settings screen slides from right to left, or
b) app settings -> privacy settings 
3. now switch to another app
4. finally try to switch back to the WP app

##### Tested on [Google Pixel], Android [8.0]

--- a/WordPress/src/main/java/org/wordpress/android/util/DialogExtensions.kt
+++ b/WordPress/src/main/java/org/wordpress/android/util/DialogExtensions.kt
@@ -6,7 +6,7 @@ import android.os.Build
 import android.view.View
 
 @SuppressLint("InlinedApi")
-fun Dialog.getContainerView(): View? {
+fun Dialog.getPreferenceDialogContainerView(): View? {
     val containerViewId = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
         android.R.id.list_container
     } else {

--- a/WordPress/src/main/java/org/wordpress/android/util/DialogExtensions.kt
+++ b/WordPress/src/main/java/org/wordpress/android/util/DialogExtensions.kt
@@ -1,0 +1,17 @@
+package org.wordpress.android.util
+
+import android.annotation.SuppressLint
+import android.app.Dialog
+import android.os.Build
+import android.view.View
+
+@SuppressLint("InlinedApi")
+fun Dialog.getContainerView(): View? {
+    val containerViewId = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+        android.R.id.list_container
+    } else {
+        android.R.id.list
+    }
+
+    return findViewById(containerViewId)
+}

--- a/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java
+++ b/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java
@@ -1,6 +1,5 @@
 package org.wordpress.android.util;
 
-import android.annotation.SuppressLint;
 import android.app.Dialog;
 import android.app.Fragment;
 import android.content.ComponentName;
@@ -31,28 +30,20 @@ public class WPActivityUtils {
             return;
         }
 
-        Toolbar toolbar;
-        if (dialog.findViewById(android.R.id.list) == null
-            && dialog.findViewById(android.R.id.list_container) == null) {
+        View dialogContainerView = DialogExtensionsKt.getContainerView(dialog);
+
+        if (dialogContainerView == null) {
             return;
         }
 
-        @SuppressLint("InlinedApi") View child = dialog.findViewById(android.R.id.list_container);
-        if (child == null) {
-            child = dialog.findViewById(android.R.id.list);
-            if (child == null) {
-                return;
-            }
-        }
-
         // find the root view, then make sure the toolbar doesn't already exist
-        ViewGroup root = (ViewGroup) child.getParent();
+        ViewGroup root = (ViewGroup) dialogContainerView.getParent();
         if (root.findViewById(R.id.toolbar) != null) {
             return;
         }
 
-        toolbar = (Toolbar) LayoutInflater.from(context.getActivity())
-                                          .inflate(org.wordpress.android.R.layout.toolbar, root, false);
+        Toolbar toolbar = (Toolbar) LayoutInflater.from(context.getActivity())
+                                                  .inflate(org.wordpress.android.R.layout.toolbar, root, false);
         root.addView(toolbar, 0);
 
         dialog.getWindow().setWindowAnimations(R.style.DialogAnimations);
@@ -85,7 +76,14 @@ public class WPActivityUtils {
             return;
         }
 
-        ViewGroup root = (ViewGroup) dialog.findViewById(android.R.id.list).getParent();
+        View dialogContainerView = DialogExtensionsKt.getContainerView(dialog);
+
+        if (dialogContainerView == null) {
+            return;
+        }
+
+        ViewGroup root = (ViewGroup) dialogContainerView.getParent();
+
         if (root.getChildAt(0) instanceof Toolbar) {
             root.removeViewAt(0);
         }
@@ -133,12 +131,12 @@ public class WPActivityUtils {
     public static void disableComponent(Context context, Class<?> klass) {
         PackageManager pm = context.getPackageManager();
         pm.setComponentEnabledSetting(new ComponentName(context, klass),
-                                      PackageManager.COMPONENT_ENABLED_STATE_DISABLED, PackageManager.DONT_KILL_APP);
+                PackageManager.COMPONENT_ENABLED_STATE_DISABLED, PackageManager.DONT_KILL_APP);
     }
 
     public static void enableComponent(Context context, Class<?> klass) {
         PackageManager pm = context.getPackageManager();
         pm.setComponentEnabledSetting(new ComponentName(context, klass),
-                                      PackageManager.COMPONENT_ENABLED_STATE_ENABLED, PackageManager.DONT_KILL_APP);
+                PackageManager.COMPONENT_ENABLED_STATE_ENABLED, PackageManager.DONT_KILL_APP);
     }
 }

--- a/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java
+++ b/WordPress/src/main/java/org/wordpress/android/util/WPActivityUtils.java
@@ -19,6 +19,7 @@ import android.view.WindowManager;
 import android.widget.TextView;
 
 import org.wordpress.android.R;
+import org.wordpress.android.util.AppLog.T;
 
 import java.util.List;
 
@@ -30,9 +31,10 @@ public class WPActivityUtils {
             return;
         }
 
-        View dialogContainerView = DialogExtensionsKt.getContainerView(dialog);
+        View dialogContainerView = DialogExtensionsKt.getPreferenceDialogContainerView(dialog);
 
         if (dialogContainerView == null) {
+            AppLog.e(T.SETTINGS, "Preference Dialog View was null when adding Toolbar");
             return;
         }
 
@@ -76,9 +78,10 @@ public class WPActivityUtils {
             return;
         }
 
-        View dialogContainerView = DialogExtensionsKt.getContainerView(dialog);
+        View dialogContainerView = DialogExtensionsKt.getPreferenceDialogContainerView(dialog);
 
         if (dialogContainerView == null) {
+            AppLog.e(T.SETTINGS, "Preference Dialog View was null when removing Toolbar");
             return;
         }
 

